drivers/flash/flash_stm32_ospi.c

```diff
@@ -955,6 +962,34 @@ static int stm32_ospi_mem_reset(const struct device *dev)
 	return 0;
 }
 
+/* Set the erase command according to erase type, SPI mode and address size */
+static void ospi_set_erase_cmd(const struct device *dev, OSPI_RegularCmdTypeDef *cmd_erase,
+			  enum flash_erase_type erase_mode)
+{
+	const struct flash_stm32_ospi_config *dev_cfg = dev->config;
+
+	if (erase_mode == SECTOR_ERASE) {
+		if (dev_cfg->data_mode == OSPI_OPI_MODE) {
+			cmd_erase->Instruction = SPI_NOR_OCMD_SE;
+		} else {
+			cmd_erase->Instruction =
+				(stm32_ospi_hal_address_size(dev) == HAL_OSPI_ADDRESS_32_BITS)
+				 ? SPI_NOR_CMD_SE_4B
+				 : SPI_NOR_CMD_SE;
+		}
+	}
+	if (erase_mode == BLOCK_ERASE) {
+		if (dev_cfg->data_mode == OSPI_OPI_MODE) {
+			cmd_erase->Instruction = SPI_NOR_OCMD_BE;
+		} else {
+			cmd_erase->Instruction =
+				(stm32_ospi_hal_address_size(dev) == HAL_OSPI_ADDRESS_32_BITS)
+				 ? SPI_NOR_CMD_BE_4B
+				 : SPI_NOR_CMD_BE;
+		}
+	}
```

Comment:
This probably should be enabled/disabled by the Kconfig.
Reason is that BLOCK erase can take ms to execute which means that device will not be accessible for that time for other operations; in case of multithreaded environment this may lock/lag/timeout threads that would attempt to access the device for read/write in other areas than erased. 
Other SPI drivers probably have the same problem.

Reply:
The function is called erase_mode == BLOCK_ERASE only when block_erase_size is set in DTS.